src/components/app-components/Page.tsx

```diff
@@ -29,12 +37,15 @@ class Page extends React.Component<PageProps> {
         return (
             <Box style={props.style} className={classNames}>
                 <Head>
-                    <link href="https://fonts.googleapis.com/css2?family=Oswald:wght@300;400;500&family=Roboto:ital,wght@0,300;0,400;0,500;1,400&display=swap" rel="stylesheet"/>
+                    <link rel="stylesheet"
+                          href="https://fonts.googleapis.com/css2?family=Oswald:wght@300;400;500&family=Roboto:ital,wght@0,300;0,400;0,500;1,400&display=swap"/>
 
-                    <title>{props.title}</title>
+                    <title>
+                        {props.title}
+                    </title>
                 </Head>
 
-                {props.children}
+                {this.state.show && props.children}
```

Comment:
If the page requires login, then it hides the children until it can verify that the user is logged in (which is also done one this page).

I'm not sure if this is the best way to do this, but it's what i've got for now. @EcZachly, let me know if we can improve this.